iron-location.html

```diff
@@ -209,14 +209,16 @@
         // nothing to do, the URL didn't change
         return;
       }
+      // Need to use a full URL in case the containing page has a base URI.
+      var fullNewUrl = new URL(newUrl, window.location.href).href;
```

Comment:
Hm, good point. What is the least surprising behavior when someone does:

```javascript
ironLocation.path = 'foo';
```

My inclination is that we should either error out or treat it the same as assigning to `/foo`. Path *was* absolute before the assignment, and it's absolute again afterwards. Furthermore, `window.location.pathname = 'foo';` redirects to `/foo`, not `./foo`.

---------

So, long story short, I agree. Switched to using the origin as the base URL, and added a test. Similar code isn't needed for the other case at the bottom of the file because that url is already absolute.